mux_test.go

```diff
@@ -1967,6 +1967,314 @@ func TestErrMatchNotFound(t *testing.T) {
 	}
 }
 
+// methodsSubrouterTest models the data necessary for testing handler
+// matching for subrouters created with set HTTP methods.
+type methodsSubrouterTest struct {
+	title      string
+	router     *Router
+	flags      []bool
+	methods    []string
+	method     string
+	path       string
+	redirectTo string
+	wantCode   int
+}
+
+// TestMethodsSubrouter tests the correct matching of handlers for
+// subrouters registered after a route's method-matcher is set.
+func TestMethodsSubrouter(t *testing.T) {
+	flags1 := make([]bool, 3)
```

Comment:
It isn't obvious (e.g. at first) what the purpose of the slice of bools / flags are for at first. Can you add a comment here to aid future contributors, and/or (better) look at a better way to test these that doesn't require this state tracking?

Reply:
Yes, I can try that, would inspecting the response body be better?

Reply:
I actually don't quite understand it, is it just to show which methods matched? Why not use a `map[string]bool` for this?

Reply:
Yes, the flags are used to check which handler is invoked. I wanted to check sequentially (for when two same routes are registered, but only the first one should be matched), but I can use `map[string]bool` with `[method][number]` like `get1` and `get2`.

Reply:
I would also call it something like `matched_methods` instead of just `flags`. The tests for this package are pretty confusing and convoluted, so anything that helps add some clarity is a big plus.